src/ATen/native/xpu/AdaptiveAveragePooling2d.cpp

```diff
@@ -104,16 +30,6 @@ Tensor XPUNativeFunctions::_adaptive_avg_pool2d_backward(
       (input.ndimension() == 3 || input.ndimension() == 4),
       "non-empty 3D or 4D (batch mode) tensor expected for input");
 
-  if (grad_output.size(-1) == 1 && grad_output.size(-2) == 1) {
```

Comment:
Pls add the dependency PR in the comments.

Reply:
https://github.com/pytorch/pytorch/pull/132217